sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -937,28 +941,10 @@ private void SetNameFieldsIfNull()
             bool async,
             CancellationToken cancellationToken)
         {
-            // TODO #27253
-            //BlobDownloadOptions options = default;
-            //if (range != default || conditions != default || rangeGetContentHash != default)
-            //{
-            //    options = new BlobDownloadOptions
-            //    {
-            //        Range = range,
-            //        Conditions = conditions,
-            //        RangeGetContentHash = rangeGetContentHash
-            //        // TODO #27253
-            //        //TransactionalHashingOptions = rangeGetContentHash
-            //        //    ? new DownloadTransactionalHashingOptions()
-            //        //    {
-            //        //        Algorithm = TransactionalHashAlgorithm.MD5
-            //        //    }
-            //        //    : default,
-            //    };
-            //}
             Response<BlobDownloadStreamingResult> response = await DownloadStreamingDirect(
                 range,
                 conditions,
-                rangeGetContentHash,
+                rangeGetContentHash.ToValidationOptions(),
```

Comment:
Does `ToValidationOptions()` handle null?  (Not sure if this matters)

Reply:
Yes

```csharp
public static UploadTransferValidationOptions ToValidationOptions(this byte[] md5)
    => md5 == default
        ? default
        : new UploadTransferValidationOptions
        {
            Algorithm = ValidationAlgorithm.MD5,
            PrecalculatedChecksum = md5
        };
```